packages/docregistry/src/default.ts

```diff
@@ -616,6 +617,21 @@ export class DocumentWidget<
     if (args.name === 'dirty') {
       this._handleDirtyState();
     }
+    if (!this.context.model.dirty) {
```

Comment:
Can you remind my what the `dirty` check does here?

Reply:
If I remember correctly if `this.context.model.dirty === true` then `this.context.contentsModel.writable` could also be true even if it was a read only file or vice-versa (I forget which). By adding this check here, I'm making sure it doesn't try to add a read only label until the context is fully initialized.